src/Components/Components/src/SupplyParameterFromPersistentComponentStateProviderServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using static Microsoft.AspNetCore.Internal.LinkerFlags;
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Enables component parameters to be supplied from <see cref="PersistentComponentState"/> with <see cref="SupplyParameterFromPersistentComponentStateAttribute"/>.
+/// </summary>
+public static class SupplyParameterFromPersistentComponentStateProviderServiceCollectionExtensions
+{
+    /// <summary>
+    /// Enables component parameters to be supplied from <see cref="PersistentComponentState"/> with <see cref="SupplyParameterFromPersistentComponentStateAttribute"/>..
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/>.</param>
+    /// <returns>The <see cref="IServiceCollection"/>.</returns>
+    public static IServiceCollection AddSupplyValueFromPersistentComponentStateProvider(this IServiceCollection services)
+    {
+        services.TryAddEnumerable(ServiceDescriptor.Scoped<ICascadingValueSupplier, SupplyParameterFromPersistentComponentStateValueProvider>());
+        return services;
```

Comment:
Would it be bad if we just registered this in DI by default (with other default Blazor services) and didn't have any public API to control it?

Reply:
It lives in Microsoft.AspNetCore.Components, we can't do it or else `SupplyParameterFromPersistentComponentStateValueProvider` needs to be public, which I don't want.

Reply:
What about having the standard hosting models call `AddSupplyValueFromPersistentComponentStateProvider` by default? Do they already do that? Just thinking of how to make this as natural and first-class as possible.

Reply:
Yes, the different host models already do that.